src/v/cluster/types.h

```diff
@@ -174,6 +174,21 @@ inline std::error_code make_error_code(tx_errc e) noexcept {
     return std::error_code(static_cast<int>(e), tx_error_category());
 }
 
+using kafka_offset = named_type<int64_t, struct kafka_offset_type>;
+struct kafka_result {
+    kafka_offset last_offset;
+};
+struct kafka_stages {
+    kafka_stages(ss::future<>, ss::future<result<kafka_result>>);
+    explicit kafka_stages(raft::errc);
```

Comment:
I would add constructor accepting only `request_enqueued` this way we wouldn't have to use `kafka_offset(errc::success)`, in all the cases the ready future with success error code is then discarded.

Reply:
fixed without introducing the new constructor by building kafka_stages just before returning it when all the arguments are already known